migrate/migrate.go

```diff
@@ -51,14 +72,25 @@ func RunMigrations(pool *sql.DB, connection string, opts MigrateOptions) error {
 		return fmt.Errorf("failed to run scripts: %w", err)
 	}
 
+	parsedConn, err := url.Parse(connection)
+	if err != nil {
+		return err
+	}
+	user := parsedConn.User.Username()
+	if user == "" {
+		return fmt.Errorf("malformed connection string, got empty username: %s", parsedConn.Redacted())
+	}
+
 	logger.Tracef("Granting roles to current user")
 	// Grant postgrest roles in ./functions/postgrest.sql to the current user
-	if err := grantPostgrestRolesToCurrentUser(pool, connection); err != nil {
+	if err := grantPostgrestRolesToCurrentUser(pool, user); err != nil {
 		return fmt.Errorf("failed to grant postgrest roles: %w", err)
 	}
 
+	setStatementTimeouts(ctx, pool, user, opts)
```

Comment:
I think the statement timeous should come after the migrations have run, especially as the timeout value could come from the properties table

Reply:
We also then wouldn't need to change the signatures of Init/Migrate